exist-core/src/main/java/org/exist/xquery/functions/util/Eval.java

```diff
@@ -438,7 +438,7 @@ private Sequence doEval(final XQueryContext evalContext, final Sequence contextS
     private void cleanup(final XQueryContext evalContext, final XQueryContext innerContext, final DocumentSet oldDocs,
             final LocalVariable mark, final Item expr, final Sequence resultSequence) {
         if (innerContext != evalContext) {
-            innerContext.reset(true);
+            evalContext.addImportedContext(innerContext);
```

Comment:
so effectively the reset is removed here.... it is reset indirect?

Reply:
Yes, but the reset happens in a different way, and the purpose of it is more explicit.